crates/orderbook/src/ipfs_app_data.rs

```diff
@@ -118,6 +119,26 @@ impl IpfsAppData {
         metric.with_label_values(&[outcome(&result), "node"]).inc();
         Ok(result)
     }
+
+    pub async fn post_app_data(&self, data: ValidatedAppData) -> Result<PinResponse> {
+        tracing::debug!("posting appData to IPFS {:?}", data);
+        self.ipfs.add(data.into()).await
+    }
+}
+
+impl From<ValidatedAppData> for PinByHashRequest {
+    fn from(value: ValidatedAppData) -> Self {
+        // Parse the string into serde_json::Value
+        let parsed_json: Value = serde_json::from_str(&value.document).unwrap();
+        // build escaped json string.
+        let escaped_json_string = serde_json::to_string(&parsed_json.to_string()).unwrap();
```

Comment:
Since already have a `ValidatedAppData` we should no longer have to parse the content to verify it's valid JSON.
Also we should use the `value.document` as is instead of deserializing and re-serializing since this might change its representation which can result in a different CID.

Reply:
Problem with document is that it's not escaped properly for ipfs api.

Reply:
Hmm, strange that they require a specific format for the content. Btw when looking into [pinByHash](https://docs.pinata.cloud/api-reference/endpoint/pin-by-cid#openapi-post-pinningpinbyhash) they say it only pins document that have already been uploaded to IPFS. It looks like what we'd actually want to use is [pinFileToIpfs](https://docs.pinata.cloud/api-reference/endpoint/pin-file-to-ipfs) to actually make pinata upload the file.
Or alternatively as discussed in some other comment using https://github.com/cowprotocol/ipfs-block-put should also work with the raw strings.

Reply:
Ya maybe I misunderstood what uploading a file to ipfs is. I guess pinning is not the same thing. Or needs to be done after. My thought was that pin by hash would upload the document and link it to our specific cid.